### PR TITLE
x265: bump to x265 v3.0

### PR DESCRIPTION
**Pull request by kenmays**
x265 HEVC Encoder / H.265 Video Codec - http://x265.org/
Built successfully on hrev53102

### CODE REVIEW

**Comment by korli** [comment]
This PR supersedes https://github.com/haikuports/haikuports/pull/3329 but the comments in the old PR still stand

File: media-libs/x265/patches/x265-3.0.patchset [hunk 1]
```diff
@@ -90,3 +80,20 @@ index f359d80..55679dd 100644
...
+diff --git a/source/CMakeLists.txt.klm b/source/CMakeLists.txt
+index bed6ff5..656c239 100644
+--- a/source/CMakeLists.txt.klm
++++ b/source/CMakeLists.txt
+@@ -558,7 +558,8 @@ if(ENABLE_HDR10_PLUS)
+         LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+         ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+ endif()
+-install(FILES x265.h "${PROJECT_BINARY_DIR}/x265_config.h" DESTINATION include)
++install(FILES x265.h "${PROJECT_BINARY_DIR}/x265_config.h"
```
[the remark this answers is not in the document]
Comment (kenmays) [reply media-libs/x265/patches/x265-3.0.patchset]: Pushing patches to dev team.

**Comment by kenmays** [comment]
Packages affected::
```
attica_x86-5.55.0-2
calligra_x86-3.1.0-3
clementine_x86-1.3.1~git-4
coolreader3_x86-3.2.2.5m-1
fbreader_x86-0.99.4-3
fotowall_x86-1.0-1
fritzing_x86-0.9.3b-1
ghostwriter_x86-1.7.3-1
goldendict_x86-1.5.0rc2-5
grantlee_x86-5.1.0-1
gst_plugins_bad_x86-1.16.0-1
kactivities_x86-5.55.0-2
karchive_x86-5.55.0-2
kate_x86-19.03.80-1
kauth_x86-5.55.0-2
kbookmarks_x86-5.55.0-2
kcmutils_x86-5.55.0-2
kcodecs_x86-5.55.0-2
kcompletion_x86-5.55.0-2
kconfig_x86-5.55.0-2
kconfigwidgets_x86-5.55.0-2
kcoreaddons_x86-5.55.0-2
kcrash_x86-5.55.0-2
kdbusaddons_x86-5.55.0-2
kdeclarative_x86-5.55.0-2
kded_x86-5.55.0-2
kdelibs4support_x86-5.55.0-2
kdevelop_x86-5.2.80-2
kdiagram_x86-2.6.0-1
kdiff3_x86-0.9.98-3
kdnssd_x86-5.55.0-2
kdoctools_x86-5.55.0-2
kemoticons_x86-5.55.0-2
kfilemetadata_x86-5.55.0-2
kglobalaccel_x86-5.55.0-2
kguiaddons_x86-5.55.0-2
kholidays_x86-5.55.0-2
khtml_x86-5.55.0-2
ki18n_x86-5.55.0-2
kiconthemes_x86-5.55.0-2
kinit_x86-5.55.0-2
kio_x86-5.55.0-2
kitemmodels_x86-5.55.0-2
kitemviews_x86-5.55.0-2
kjobwidgets_x86-5.55.0-2
kjs_x86-5.55.0-2
knewstuff_x86-5.55.0-2
knotifications_x86-5.55.0-2
knotifyconfig_x86-5.55.0-2
kolourpaint_x86-19.03.80-4
kpackage_x86-5.55.0-2
kparts_x86-5.55.0-2
kplotting_x86-5.55.0-2
kproperty_x86-3.1.0-1
krdc_x86-18.08.0-1
kreport_x86-3.1.0-1
krita_x86-4.1.0-2
kross_x86-5.55.0-2
kservice_x86-5.55.0-2
kstars_x86-2.9.8-2
ksyntax_highlighting_x86-5.55.0-2
ktexteditor_x86-5.55.0-2
ktextwidgets_x86-5.55.0-2
kunitconversion_x86-5.55.0-2
kwallet_x86-5.55.0-2
kwave_x86-19.03.80-1
kwidgetsaddons_x86-5.55.0-2
kwindowsystem_x86-5.55.0-2
kxmlgui_x86-5.55.0-2
libkomparediff2_x86-17.12.0-2
libksane_x86-19.03.80-1
libopenshot_x86_python3-0.2.3-1
libopenshot_x86-0.2.3-1
librecad_x86-2.1.3-5
libreoffice_x86-6.2.4.1-1
lmms_x86-1.2.0~rc8-1
musescore_x86-3.0.0_git-1
openshot-2.4.4-2
otter_browser_x86-1.0.01-7
pencil_x86-0.6.2-1
phonon_gstreamer_x86-4.9.0-3
phonon_x86-4.10.2-2
poppler_x86_qt5-0.62.0-1
pyqt_x86_python3-5.12.1-2
pyqt_x86-5.12.1-2
qca_x86-2.2.1-1
qcad_x86-3.21.3.0-1
qelectrotech_x86-0.6-2
qmmp_x86-1.3.1-2
qmplay2_x86-18.12.26-4
qputty_x86-506-1
qt_creator_x86-4.8.1-2
qt5_x86_devel-5.12.3-1
qt5_x86_tools-5.12.3-1
qt5_x86-5.12.3-1
qthaikuplugins_x86-5.12.3.4-1
qtwebkit_x86-5.212.0~pre20180120-7
qvim_x86-8.0.197-1
scribus_x86-1.5.4-2
smplayer_x86-19.1.0-1
smtube_x86-18.11.0-1
solid_x86-5.55.0-2
sonnet_x86-5.55.0-3
tbe_x86-0.9.3.1-1
threadweaver_x86-5.55.0-2
```

**Comment by korli** [comment]
From @alaviss 
> Note that there are recipes that uses `libx265`, and should have their `REVISION` bumped. Since `libx265` is a quite popular library, we might have to make it co-installable to avoid breaking ABI.

File: media-libs/x265/patches/x265-3.0.patchset (hunk 1 above)
[the remark this answers is not in the document]
Comment (extrowerk) [reply media-libs/x265/patches/x265-3.0.patchset]: I can't seem to find your patch anywhere.

**Comment by kenmays** [comment]
@alaviss - I created a list of all the deps. Can rebuild after merge, but we should do only if needed.

**Comment by kenmays** [comment]
We currently use x265 2.0-2. Any x265 ABI-backward compatibility issues known?

**Pull request merged**